base weapon recoil generation change+ small bugfix + hybrid trace changes

LeninCat opened the pull request:
-Removed intense screenshake from aimpunch produced by high latency
-Added client initiated muzzle climb recoil
-Fixed recoil and aimpunch being applied while sprinting

LeninCat committed:
base weapon recoil generation change+ small bugfix

-Removed intense screenshake from aimpunch produced by high latency
-Added client initiated muzzle climb recoil
-Fixed recoil and aimpunch being applied while sprinting

LiddulBOFH reviewed (changes requested):
May want to still keep Ply as a localized version of the player, calling self.Owner repeatedly is not good for performance due to the extremely inefficient metatable index entities have, and self:GetOwner() is preferred over self.Owner
"IFTP" is also not needed as it is only being used once, not to mention its being defined in the global space when it should be localized
"ang" should also be localized

LeninCat committed:
changes requested has been applied

requested changes to base weapon code from pr requested has been applied

LeninCat commented:
changes has been applied

LeninCat committed:
missed line

missed a line

LeninCat committed:
base weapon hybrid trace changes+ linterhell

-Hybrid trace now only initiates when hitting Player or NPC
-Added selectable effect to hybrid trace. BloodImpact and ACF_Penetration.
-linterhell (a.)

LeninCat commented:
-hybrid trace changes added (effects added)
-linterhell

LeninCat commented:
any update on merge?

LiddulBOFH commented:
Sorry, it's been a hot minute since I've really sat down to do things. For the most part it looks good, just one of those micro optimizations, that IFTP variable still bugs me particularly when its defined and used a single time like that, given how conditionals work it'd be better in this case to call the function where it's needed instead of setting it in memory and then reading it. I'll merge and make a few other changes before updating workshop.

LeninCat commented:
Thanks for the merge, as for the IFTP i had to keep it the way it is to prevent the eye angle recoil from continuously applying more than intended when the player has more ping than like 20. There might be another way to not use this method to prevent the eye angle creep but i don't know how.
